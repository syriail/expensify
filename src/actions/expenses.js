import uuid from 'uuid';
import database from '../firebase/firebase';

export const addExpense = (expense) => ({
    type: 'ADD_EXPENSE',
    expense
});

export const startAddExpense = (expenseData = {})=>{
    return (dispatch)=>{
        const {
            description = '', note = '', amount = 0, createAt = 0
        } = expenseData;

        const expense = {description, note, amount, createAt};
        console.log(expense);
        //We return here to chain another then in test cases. See expenses.test.js
        return database.ref('expenses').push(expense).then((ref)=>{
            dispatch(addExpense({
                id: ref.key,
                ...expense
            }));
        });
    };
};

export const removeExpense = ({id} = {}) => ({
    type: 'REMOVE_EXPENSE',
    id
});

export const editExpense = (id, updates) => ({
    type: 'EDIT_EXPENSE',
    id,
    updates
});

