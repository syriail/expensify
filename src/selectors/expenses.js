import moment from 'moment';

export default (expenses, {text, sortBy, startDate, endDate})=>{
    var visibleExpenses = expenses.filter(({description, createAt}) =>{
        const createAtMoment = moment(createAt);
        const textMatch = description.toLowerCase().includes(text.toLowerCase());
        const startDateMatch = startDate ? startDate.isSameOrBefore(createAtMoment, 'day') : true;
        const endDateMatch = endDate ? endDate.isSameOrAfter(createAtMoment, 'day') : true;
        return textMatch && startDateMatch && endDateMatch;
    }).sort((a, b)=>{
        if(sortBy === 'date'){
            return a.createAt < b.createAt ? 1 : -1;
        }else{
            return a.amount < b.amount ? 1 : -1;
        }
    });
    return visibleExpenses;
};