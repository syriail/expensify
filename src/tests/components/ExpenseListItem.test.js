import React from 'react';
import {shallow} from 'enzyme';
import expenses from '../fixtures/expenses';
import ExpenseListItem from '../../components/ExpenseListItem';

test('Should render Expense list item correctly', ()=>{
    const wrapper = shallow(<ExpenseListItem {...expenses[2]}/>);
    expect(wrapper).toMatchSnapshot();
});

