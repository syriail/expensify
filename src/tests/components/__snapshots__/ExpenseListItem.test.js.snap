// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Expense list item correctly 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/3"
  >
    <h3>
      Gas bill
    </h3>
  </Link>
  <p>
    1250
     - 
  </p>
</div>
`;
